# **Step 1**: **Read** the caption script text

# In this tutorial, we read the script (variable = 'combine_srt') from a url on Github; therefore we use the requests library
### Otherwise, if we read the script locally, we'll need 'Path' in the 'pathlib' library to read the actual .srt file
### from pathlib import Path 

In [60]:
import requests as re

# **Assigned** the variable, 'combine_srt', to hold the script url 

In [61]:
combine_srt = r'https://raw.githubusercontent.com/clueple/free_resources/master/combine_srt.srt'

# **Turn** the 'combine_srt' (a url) to a text file and then apply a string function **'splitlines'** to **make it a list**

In [62]:
combine = re.get(combine_srt).text.splitlines()

# Preview the type of the variable, 'combine'

In [63]:
type(combine)

list

# Preview the variable, 'combine' (only the first 16 lines, 3 lines of caption)

In [64]:
combine [0:15]

['1',
 '00:00:00,000 --> 00:00:01,033',
 'In this video ',
 '喺呢條片',
 '',
 '2',
 '00:00:01,033 --> 00:00:04,500',
 "we'll learn the 3 basic steps on building apps ",
 '我哋會學整app嘅3個基本步驟',
 '',
 '3',
 '00:00:04,500 --> 00:00:05,833',
 "We call it 'ADD' ",
 "叫做 'ADD'",
 '']

# Our target is to extract the English text ('top' language) and Cantonese ('bottom' language) respectively

# Let's bring up some terms that we're going to use in our code.
# Pick the **first line** as an example:

'1',\
 '00:00:00,000 --> 00:00:01,033',\
 'In this video ',\
 '喺呢條片'
 
 where '1' is the **index** \
 '00:00:00,000 --> 00:00:01,033' is the **timestamp** \
 'In this video ' is the **'top' language** 
 
 # if we extract the 'top' language, we expect to get: 
 '1',\
 '00:00:00,000 --> 00:00:01,033',\
 'In this video ',
 

# **In caption index 1**, 
'i', the current line \
line 0 is '1', the  **'index'** \
line 1 is '00:00:00,000 --> 00:00:01,033', the **timestamp** \
line 2 is 'In this video', the **content**
line 3 is '' , the **space**

so for **each caption index** ...


In [65]:
# the caption index will row over to the next one every 5 lines (e.g. index 1 to index 2)
step = 5 

# the beginning position of the 'top' language
begin_pos_top = 2

# for each line in the list, 'combine'
for i in range(0, len(combine)):
    if i % step == begin_pos_top:
        index_top = i - 2
        timestamp_top = i - 1
        content = f"{combine[index_top]}\n{combine[timestamp_top]}\n{combine[i]}\n"
        print(content)


1
00:00:00,000 --> 00:00:01,033
In this video 

2
00:00:01,033 --> 00:00:04,500
we'll learn the 3 basic steps on building apps 

3
00:00:04,500 --> 00:00:05,833
We call it 'ADD' 

4
00:00:06,133 --> 00:00:08,500
appearance, details, and debug 

5
00:00:08,633 --> 00:00:09,200
Here 

6
00:00:09,200 --> 00:00:14,033
we'll use the simple YouTube Downloader app as an example 

7
00:00:14,166 --> 00:00:16,632
First of all, we'll deal with the look 

8
00:00:16,633 --> 00:00:19,300
That'll be handled in the layout section 

9
00:00:19,400 --> 00:00:20,532
You're very lucky 

10
00:00:20,533 --> 00:00:24,900
coz you can download a ready-to-view template on Github 

11
00:00:24,900 --> 00:00:27,300
If you wanna dig deep on how to do it 

12
00:00:27,366 --> 00:00:30,432
feel free to watch the episode 2 and 3 

13
00:00:30,866 --> 00:00:33,799
Once you've designed the app's layout 

14
00:00:33,800 --> 00:00:35,867
you need to fill in the details 

15
00:00:36,033 --> 00:00:38,267
In our YouTub